rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,341 @@
+---
+title: 'Large Message Handling in GossipSub: Potential Improvements'
+date: 2024-06-24 15:00:00
+authors: farooq
+published: false
+slug: GossipSub Improvements
+categories: research
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+Large Message Handling in GossipSub: Potential Improvements 
+
+<!--truncate-->
+
+## Motivation:
+The challenge of large message transmissions in gossipsub leads to longer than expected network-wide message dissemination times (and relatively higher fluctuations). 
+It is particularly relevant for applications like Waku and Ethereum [], 
+that require on-time network-wide dissemination of large messages. 
+
+This matter has been extensively discussed in the libp2p community [], 
+and numerous improvements have been suggested (or even incorporated) to the gossipsub protocol to enable efficient large-message propagation.
+
+## Problem Realization:
+
+Sending a message to N peers involves approximately $\lceil \log_D(N) \rceil$ transmission rounds, 

```

Comment:
Perhaps simply say 'rounds' instead of 'transmission rounds'?

Reply:
bd3f3d9c0f32350e07e987ff0ffc46ed9e3ba174